hs-ucan/library/Web/UCAN/Types.hs

```diff
@@ -314,3 +327,15 @@ signRS256 header claims sk =
   RSA.PKCS15.signSafer (Just SHA256) sk (encodeUtf8 $ B64.URL.encodeJWT header claims) <&> \case
     Left err  -> Left err
     Right sig -> Right . Signature.RS256 $ RS256.Signature sig
+
+signSecp256k1 :: ( ToJSON fct
+  , ToJSON rsc
+  , ToJSON ptc
+  )
+  => Header
+  -> Claims fct rsc ptc
+  -> Secp256k1.SecKey
+  -> Maybe Signature.Signature
+signSecp256k1 header claims sk =
+  -- Only works if the signed data is 32 bytes
+  Signature.Secp256k1 . Secp256k1.Signature . Secp256k1.exportRecoverableSignature <$> Secp256k1.ecdsaSignRecoverable sk (encodeUtf8 $ B64.URL.encodeJWT header claims)
```

Comment:
Need to take a closer look at this. The way we sign things with Metamask/Ethereum is not the same as this. This doesn't have the prefix and keccak256 hashing.

Reply:
It does test them automatically. At least it generates UCANs with secp256k1 keys:
https://github.com/fission-codes/fission/blob/221d1f3189a90b8a60c9d1d42e35d62c312bb900/hs-ucan/library/Web/UCAN/Types.hs#L108-L114

Reply:
Having an `Arbitrary Algorithm` that adds `Secp256k1` as one of the options leads to `Arbitrary UCAN.Header` to sometimes generate with secp256k1 as the algorithm which in turn causes the code I linked above to generate secp256k1 signatures.

Reply:
Ah awesome 👏 I guess that plus the DID tests covers everything.